File: carcassonne-master/src/bottom_side_connections.cc
#include "bottom_side_connections.h"

namespace carcassonne
{

BottomSideConnections::BottomSideConnections() : SideConnections()
{
}

BottomSideConnections::BottomSideConnections(const BottomSideConnections& src)
	: SideConnections(src)
{
	CopyFrom(src);
}

BottomSideConnections& BottomSideConnections::
	operator=(const BottomSideConnections& rhs)
{
	if(this == &rhs) {
		return (*this);
	}
	
	CopyFrom(rhs);
	
	return (*this);
}

void BottomSideConnections::
	CopyFrom(const BottomSideConnections& src)
{										  
}

SideConnections* BottomSideConnections::
	Clone() const
{
	return new BottomSideConnections(*this);
}

std::vector<std::string> BottomSideConnections::ToStringVector() const
{
  std::vector<std::string> connections;

  connections.push_back("Bottom(itself)");

  if(clockwise_) {
    connections.push_back("Left");
  }

  if(across_) {
    connections.push_back("Top");
  }

  if(counterclockwise_) {
    connections.push_back("Right");
  }

  return connections;
  
}

BottomSideConnections::~BottomSideConnections()
{
}

}

File: js/BottomSideConnections.js
const SideConnections=require( "./SideConnections");

module.exports= class BottomSideConnections extends SideConnections {
    constructor(src) {
        super(src);
        if (src) {
            this.CopyFrom(src);
        }

    }

    Clone() {
        return new BottomSideConnections(this);
    }

    ToStringVector() {
        var connections = [];

        connections.push("Bottom(itself)");

        if (this.clockwise_) {
            connections.push("Left");
        }

        if (this.across_) {
            connections.push("Top");
        }

        if (this.counterclockwise_) {
            connections.push("Right");
        }

        return connections;

    }
}